sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/evaluate/NamedVIPEvaluationStageTest.java

```diff
@@ -45,4 +49,40 @@ public void testDiscoveryInfoPopulated() throws Exception {
         Assert.assertTrue(vipLabel.getKey().startsWith("VIP_"));
         Assert.assertEquals(vipLabel.getValue(), "test-vip:80");
     }
+
+    @Test
+    public void testVIPIsReused() throws InvalidRequirementException {
```

Comment:
`NamedVIPEvaluationStage` will look to see if a VIP for the port/name pair in question already exists on the task's `DiscoveryInfo` by checking its labels. This will be the case when we're evaluating an existing offer requirement, since that offer requirement is created from the task definitions that were retrieved from the state store. The bug was that `NamedVIPEvaluationStage` was looking for the relevant VIP label on `DiscoveryInfo` itself, but VIP labels are actually on each `Port`, so every time an existing offer requirement was evaluated it would create a new port since it didn't find anything in `DiscoveryInfo.getLabels()` that matched what it expected.